### PR TITLE
Add badge groups to old assignment api

### PR DESCRIPTION
**Pull request by slomo**
Fixes that the existing badge UI can't be used to assign batches from groups.

### CODE REVIEW

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts [hunk 1]
```diff
@@ -87,14 +91,51 @@ export var bindPath = (
...
+
+    adhPermissions.bindScope(scope, scope.poolPath, "rawOptions", {importOptions: false});
+    scope.$watch("rawOptions", (rawOptions) => {
+        var requestBodyOptions = AdhUtil.deepPluck(rawOptions, [
+            "data", "POST", "request_body"
+        ]);
+
+        var badgeBodyOptions = _.find(
+            requestBodyOptions,
+            (body : any) => body.content_type === RIBadgeAssignment.content_type);
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts]: ` : any` is implicit and should be omitted

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts [hunk 2]
```diff
@@ -87,14 +91,51 @@ export var bindPath = (
...
+
+        var assignableBatchPaths : string = AdhUtil.deepPluck(badgeBodyOptions, [
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts]: type is probably `string[]`

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts [hunk 3]
```diff
@@ -87,14 +91,51 @@ export var bindPath = (
...
+            "data", SIBadgeAssignment.nick, "badge"
+        ]);
+
+        var promise = $q.all(_.map(assignableBatchPaths, (b) => adhHttp.get(b))).then((result) => {
+            scope.badges = _.map(result, getBadge);
+            return scope.badges;
+        });
+
+        promise.then((badges) => {
+            var groupPaths = _.union.apply(_, _.map(badges, (badge : any) => badge.groups));
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts]: implicit any

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts [hunk 4]
```diff
@@ -87,14 +91,51 @@ export var bindPath = (
...
+            groups: badge.data[SIBadge.nick].groups
         };
     };
 
+    var getGroup = (group) => {
+        return {
+            title: group.data[SITitle.nick].title,
+            path: group.path
+        };
+    };
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts]: I would expect a function called `getSomething()` to do HTTP. I know this is consistent with `getBadge()` though.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts [hunk 5]
```diff
@@ -87,14 +91,51 @@ export var bindPath = (
...
+        var assignableBatchPaths : string = AdhUtil.deepPluck(badgeBodyOptions, [
+            "data", SIBadgeAssignment.nick, "badge"
+        ]);
+
+        var promise = $q.all(_.map(assignableBatchPaths, (b) => adhHttp.get(b))).then((result) => {
+            scope.badges = _.map(result, getBadge);
+            return scope.badges;
+        });
+
+        promise.then((badges) => {
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts]: You should use chaining instead of defining a `promise` variable.

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts [hunk 6]
```diff
@@ -87,14 +91,51 @@ export var bindPath = (
...
+            var groupPaths = _.union.apply(_, _.map(badges, (badge : any) => badge.groups));
+
+            $q.all(_.map(groupPaths, (g :  any) => adhHttp.get(g))).then((result) => {
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Badge.ts]: implicit any

File: src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Wrapper.html [hunk 7]
```diff
@@ -15,21 +15,19 @@
...
-            data-on-submit="submit"
-            data-on-cancel="cancel"></adh-badge-assignment-create>
-    </div>
+    <adh-badge-assignment-create
+       data-ng-if="data.selected === 'new'"
+       data-badgeable-path="{{badgeablePath}}"
+       data-pool-path="{{poolPath}}"
+       data-show-description="showDescription"
+       data-on-submit="submit"
+       data-on-cancel="cancel"></adh-badge-assignment-create>
```
Comment (xi) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Wrapper.html]: Why did you remove the `ready`?

Comment (slomo) [inline src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/Badge/Wrapper.html]: Because of stuiped.

**Comment by xi** [comment]
#2312 is still an issue, but this already is an improvement.
